src/textual/widgets/_data_table.py

```diff
@@ -418,7 +444,7 @@ def refresh_cell(self, row_index: int, column_index: int) -> None:
         region = region.translate(-self.scroll_offset)
         self.refresh(region)
 
-    def _get_row_renderables(self, row_index: int) -> list[RenderableType]:
+    def _get_row_renderables(self, row_index: int) -> Sequence[RenderableType]:
```

Comment:
This actually failed strict type checking (`list` is contravariant, which `Sequence` is covariant, so when returning strings the original code fails typecheck).